### DISCUSSION TITLE
Why is the divisible pad part of the transforms, not the model (e.g. Unet)?

### DISCUSSION THREAD

**Discussion by john-zielke-snkeos** [Project-MONAI/MONAI#6705] [General]
If I am using a Unet or similar architecture, I need my inputs to be divisible by 2**network_depth. In order to ensure this, the current suggested solution seems to be to use monai.transforms.DivisiblePad. I was wondering if there is any reason not to implement this step directly in the model by:
- padding at the first layer to the right size (the required dimensions can then be read directly from the network itself)
- After the last layer unpadding again to the original size.

This seems to me like a better separation of concerns, since the padding seems like something specific to the model, not the data pipeline.

I would be happy to hear your opinions on why this might be a good or bad idea.
PS: To clarify, I am not trying to suggest we should change this everywhere, just wanted to know why this problem was solved this way and if the padding inside the model could lead to potential problems.

**Comment by KumoLiu** [comment]
Hi @john-zielke-snkeos, thank you for sharing your opinions. As you described, it is possible to ensure the size that is divisible by a factor of 2**network_depth directly in the model, such as `torch.pad`, `nn.conv`. However, there are a few factors that need to be considered when deciding whether to implement padding within the model or as a separate preprocessing step, just my personal opinions.

- From the design perspective, by using a separate preprocessing step such as `monai.transforms.DivisiblePad`, you can explicitly decouple the padding logic from the model architecture. This can make the codebase more modular and easier to maintain. 
- From an efficiency perspective, a separate preprocessing will only be applied once, however, resizing within the model architecture will be applied during the whole training and inference stages.

Considering the generalizability of the MONAI framework, I think using a separate resizing process is more appropriate for MONAI's implementation.
Thanks!
